cpp/include/raft/neighbors/detail/cagra/cagra_search.cuh

```diff
@@ -173,6 +147,145 @@ void search_main(raft::resources const& res,
             topk,
             set_offset(sample_filter, qid));
   }
+}
+
+template <class T,
+          class DatasetT,
+          class DatasetIdxT,
+          class InternalIdxT,
+          class DistanceT,
+          class CagraSampleFilterT>
+void lauch_vpq_search_main_core(
```

Comment:
The max dataset dim for cagra-q search shall be documented. Also might make sense to refuse building for index combinations that will not work.

Reply:
I updated the VPQ similarity calculation to fix a bug. This fix also allows us to support dim >= MAX_BLOCK_DIM.